libknet/tests/test-common.c

```diff
@@ -445,18 +445,27 @@ int knet_handle_stop(knet_handle_t knet_h)
 	return 0;
 }
 
-static int _make_local_sockaddr(struct sockaddr_storage *lo, uint16_t offset, int family)
+static int _make_local_sockaddr(struct sockaddr_storage *lo, int offset, int family)
 {
 	uint32_t port;
 	char portstr[32];
 
-	/* Use the pid if we can. but makes sure its in a sensible range */
-	port = (uint32_t)getpid() + offset;
-	if (port < 1024) {
-		port += 1024;
-	}
-	if (port > 65536) {
-		port = port & 0xFFFF;
+	if (offset < 0) {
+		/*
+		 * api_knet_link_set_config needs to access the API directly, but
+		 * it does not send any traffic, so it´s safe to ask the kernel
+		 * for a random port.
+		 */
+		port = 0;
+	} else {
+		/* Use the pid if we can. but makes sure its in a sensible range */
+		port = (uint32_t)getpid() + offset;
+		if (port < 1024) {
+			port += 1024;
+		}
+		if (port > 65536) {
+			port = port & 0xFFFF;
+		}
```

Comment:
This can leave you with any 0 <= port <= 65536, which is wrong on both ends. How about this?
```suggestion
		/* Use the pid if we can. but makes sure its in a sensible range */
		port = (getpid() + offset) % (65536-1024) + 1024;
```

Reply:
make sense to me, i´ll wait for @chrissie-c to add her review and I´ll change it.